### PR TITLE
fix: correctly handle schema check composition skip

### PR DESCRIPTION
**Pull request by n1ru4l**
### Background

An error popped up within our error monitoring system regarding this code

### Description

We made the wrong assumption that a skipped schema check is always successful. This resolves it by determining the schema check result based on the schema version we compared against.

### Checklist

- [ ] Input validation
- [ ] Output encoding
- [ ] Authentication management
- [ ] Session management
- [ ] Access control
- [ ] Cryptographic practices
- [ ] Error handling and logging
- [ ] Data protection
- [ ] Communication security
- [ ] System configuration
- [ ] Database security
- [ ] File management
- [ ] Memory management
- [ ] Testing

### CODE REVIEW

File: packages/services/api/src/modules/schema/providers/schema-publisher.ts [hunk 1]
```diff
@@ -518,51 +525,7 @@ export class SchemaPublisher {
...
-        }
-
-        if (latestSchemaVersion.compositeSchemaSDL) {
-          composition = {
-            compositeSchemaSDL: latestSchemaVersion.compositeSchemaSDL,
-            supergraphSDL: latestSchemaVersion.supergraphSDL,
-          };
-        } else {
-          // LEGACY CASE if the schema version record has no sdl
-          // -> we need to do manual composition
```
Comment (n1ru4l) [inline packages/services/api/src/modules/schema/providers/schema-publisher.ts:540]: this case did not make sense as schema versions always have SDL if composable. If not composable we should not attempt to compose...

I mixed up schema versions with schema checks when I implemented this as legacy schema check records have no SDL saved.

In addition to that I made the mistake of assuming that `SchemaCheckConclusion.Success` that was skipped is always composable...

Comment (n1ru4l) [reply packages/services/api/src/modules/schema/providers/schema-publisher.ts:540]: actually... we have legacy schema version records without SDL, the above still needs to happen in some cases... 😮‍💨

Comment (n1ru4l) [reply packages/services/api/src/modules/schema/providers/schema-publisher.ts:540]: handled in https://github.com/kamilkisiela/graphql-hive/pull/3932/commits/396f348fea1aee9cf62eabdb56f626f946db85fd

File: packages/services/api/src/modules/schema/providers/schema-publisher.ts [hunk 2]
```diff
@@ -600,13 +565,87 @@ export class SchemaPublisher {
             safeSchemaChanges: contract.schemaChanges?.safe ?? null,
           })) ?? null,
       });
+    } else if (checkResult.conclusion === SchemaCheckConclusion.Skip) {
+      if (!comparedVersion || !comparedSchemaVersion) {
+        throw new Error('This cannot happen :)');
```
Comment (kamilkisiela) [inline packages/services/api/src/modules/schema/providers/schema-publisher.ts:570]: "Hey Siri, remind me about this comment in 2 months from now"

Comment (n1ru4l) [reply packages/services/api/src/modules/schema/providers/schema-publisher.ts:570]: Either this or refactor everything and pull the checksum check out of the orchestrator in order to please TypeScript 😇

**Review by dotansimha — APPROVED** [review]
LGTM, but let's wait for @kamilkisiela approval

**Review by kamilkisiela — APPROVED** [review]

**Pull request merged**